xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -152,19 +152,34 @@ public void onListenerChanged(XdsClient.ListenerUpdate update) {
           public void onResourceDoesNotExist(String resourceName) {
             logger.log(Level.WARNING, "Resource {0} is unavailable", resourceName);
             curListener = null;
-            reportError(Status.NOT_FOUND.withDescription(resourceName).asException());
+            reportError(Status.NOT_FOUND.asException(), true);
```

Comment:
> ... boolean is determined by the Status.

Not entirely. The `XdsClient.ListenerWatcher` events are generated by the `XdsClient` (`ServerXdsClient`) in this case which has `onResourceDoesNotExist` and `onError` as 2 separate callbacks. On downstream side the last consumer is `XdsServingStatusListener` which just wants to know `onNotServing(Throwable throwable)`. Instead of assuming that boolean is determined by the Status (or the Throwable) it is best to keep these 2 things as separate and let `XdsClientWrapperForServerSds.isResourceAbsent` decide when to generate a resource absent "signal". I did go back and forth on this but then decided on this as the best way to keep the logic separate